R/toolISOhistorical.R

```diff
@@ -255,47 +255,52 @@ toolISOhistorical <- function(m, mapping = NULL, additional_mapping = NULL, over
-                                                                        bilatMapping$fromISO, paste, sep = ".")))
+          toISO <- as.vector(outer(getItems(z, dim = secdDim), bilatMapping$toISO, paste, sep = "."))
+          fromISO <- as.vector(outer(getItems(z, dim = secdDim), bilatMapping$fromISO, paste, sep = "."))
+          bilatMapping <- data.frame(toISO, fromISO)
         }
+        # subset mapping to specific transitions in the loop
+
+        # countries in transition object
+        isoTr <- getItems(z[setNames(list(c(tr$fromISO)), mainDimName), , ], dim = secdDimName)
+        wTr <- weight[setNames(list(isoTr), secdDimName), , ]
+
+        mappingTr <- bilatMapping[is.element(bilatMapping$toISO, getItems(wTr, dim = 1)),
+                                  c("fromISO", "toISO")]
 
         # make a second weight to add to original weight in cases where weight is 0 in all new countries.
         # This weight is very small value so should not affect actual values. Weight2 is ratio of each country's values
         # as ratio of sum across  partner countries as well as products.
-        weight2 <- weight
+        weight2 <- wTr
         for (i in getItems(weight2, dim = mainDim)) {
           weight2[i, , ] <- dimSums(weight2[i, , ], dim = c(secdDim, 3)) /
-                            dimSums(weight2[i, , ], dim = c(1, 3), na.rm = TRUE)
+            dimSums(weight2[i, , ], dim = c(1, 3), na.rm = TRUE)
         }
         weight2[is.na(weight2)] <- 0
         weight2 <- weight2 / 1e10
 
         zTr <- toolAggregate(z[setNames(list(c(tr$fromISO)), mainDimName), subTime, ],
-                             bilatMapping[is.element(bilatMapping$toISO,
-                                                     getItems(weight[setNames(list(c(getItems(z[setNames(list(c(tr$fromISO)), #nolint
-                                                                                                         mainDimName), , ], dim = secdDimName))),  # nolint
-                                                                              secdDimName), , ],
-                                                              dim = 1)),
-                                          c("fromISO", "toISO")],
+                             mappingTr,
                              from = "fromISO", to = "toISO",
                              dim = 1,
-                             weight = weight[setNames(list(c(getItems(z[setNames(list(c(tr$fromISO)), mainDimName), , ],
-                                                                      dim = secdDimName))), secdDimName), , ] +
-                               weight2[setNames(list(c(getItems(z[setNames(list(c(tr$fromISO)), mainDimName), , ],
-                                                                   dim = secdDimName))), secdDimName), , ], # nolint #add small weight to disaggregate properly
+                             weight = wTr + weight2, # nolint #add small weight to disaggregate properly
```

Comment:
```suggestion
                             weight = wTr + weight2, # add small weight to disaggregate properly
```
`# nolint` should not be necessary anymore, right?

Reply:
Weight 1 no longer relevant now that there is an internal function